### DISCUSSION TITLE
Search all existing mindmaps

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#806] [Ideas]
I recently discovered that I can do a search on all open mindmaps. This is a gamechanger for me because it allows me to break up huge maps into smaller maps, thus reducing the load that any one mindmap has to carry - both in terms of computer resources and in terms of putting all my eggs in one basket. If a small mindmap gets corrupted somehow, it's not nearly as big a deal as if one huge central mindmap does. I've now set my huge mindmap to read only status and have started working with smaller mindmaps.

My request is that FP be empowered to search through all existing mindmaps that are stored relative to a specified root directory, whether they are opened or not. If a match is found in an unopened file, FP would open it and go to the appropriate node.

I'm guessing this is not a trivial request. But I make it anyway in case it is seen as something to seriously consider.

~ Bal

**Comment by sifran-github** [comment]
I know that its not what you wanted, but as a workaround you can use regular programs to search for a text inside files.
Since FP stores the information in an xml file, which is basically a text file with a structure, you can search for text, and after you locate the file, just open it in FP and search regularly inside FP.

This is what I do when I want to find some node and I dont remember in which mindmap it is.

Many programs allow you to search inside files, for example notepad++

**Reply by wordmuse** [reply]
Thank you - I already to that. :) 

What I am proposing would allow me to go directly to the nodes that match the text and then quickly link to and/or from them relative to my current node.

~ Bal

**Reply by ldmpub** [reply]
@sifran-github your proposal request to performs at least 2 searches: one in the OS among files ... and one after opening the FP file.

Suggestion of @wordmuse is accurate according to me.
